eman2/protocols/protocol_tomo_template_match.py

```diff
@@ -0,0 +1,198 @@
+        form.addParam('dthr', FloatParam, default=-1,
+                      label='Distance threshold',
+                      help='')
+        form.addParam('vthr', FloatParam, default=2.0,
+                      label='Value threshold',
+                      help='')
+        form.addParam('delta', FloatParam, default=30.0,
+                      label='Delta angle',
+                      help='')
+        form.addParam('sym', StringParam, default='c1',
+                      label='Point-group symmetry',
+                      help='')
+        form.addParam('boxSize', IntParam, default=-1, label='Box size')
+
+        form.addParallelSection(threads=1, mpi=1)
+
+    # --------------------------- INSERT steps functions ----------------------
+
+    def _insertAllSteps(self):
+        self._insertFunctionStep('tempMatchStep')
+        self._insertFunctionStep("createOutputStep")
+
+
+    # --------------------------- STEPS functions -----------------------------
+
+    def tempMatchStep(self):
+
+        if self.boxSize.get()==-1:
+            self.box = self.ref.get().getDim()[0]
+        else:
+            self.box = self.boxSize.get()
+
+        volFile = os.path.abspath(self.ref.get().getFileName())
+        params = ""
+
+        for tomo in self.inputSet.get():
+            params = params + " %s" % tomo.getFileName()
+
+        params = params + " --reference=%s --nptcl=%d --dthr=%f --vthr=%f --delta=%f --sym=%s " \
+                          "--rmedge --rmgold --boxsz=%d" % (volFile, self.nptcl.get(), self.dthr.get(),
+                          self.vthr.get(), self.delta.get(), self.sym.get(), self.box)
+
+        program = eman2.Plugin.getProgram("e2spt_tempmatch.py")
+
+        self.runJob(program, params, cwd=os.path.abspath(self._getTmpPath()),
+                    env=eman2.Plugin.getEnviron())
+
+        #Move output files to Extra Path
+        moveFile(self._getTmpPath("ccc.hdf"),self._getExtraPath("particles" + ".hdf"))
+
+        for tomo in self.inputSet.get():
+            tomoName = os.path.basename(tomo.getFileName())
+            tomoName = os.path.splitext(tomoName)[0]
+            tomoCoord = "extra-" + tomoName + "_info.json"
+            moveFile(self._getTmpPath(os.path.join("info", tomoCoord)),
+                     self._getExtraPath("extra-" + tomoName + "_info.json"))
+
+        cleanPath(self._getTmpPath())
+
+    # --------------------------- INFO functions ------------------------------
+    def _validate(self):
+        errors = []
+        return errors
+
+    # --------------------------- UTILS functions ------------------------------
+    def getSummary(self, coord3DSet):
+        summary = []
+        summary.append("Number of particles picked: %s" % coord3DSet.getSize())
+        summary.append("Particle size: %s" % coord3DSet.getBoxSize())
+        return "\n".join(summary)
+
+    def _readSetOfCoordinates3D(self, jsonBoxDict, coord3DSetDict, inputTomo):
```

Comment:
Maybe this could be included in "convert" to be reused in other protocols if needed, what do you think @adrianq ?

Reply:
Yes, I agree 100% with @EstrellaFG If you extract this method to convert keep in mind exactly the same code is used in tomo boxing and therefore, you will have to change it there as well.